common/lib/dependabot/pull_request_updater/github.rb

```diff
@@ -226,13 +241,7 @@ def update_branch(commit)
         return nil if e.message.match?(/Reference does not exist/i)
         return nil if e.message.match?(/Reference cannot be updated/i)
 
-        if e.message.match?(/protected branch/i) ||
-           e.message.match?(/not authorized to push/i) ||
-           e.message.include?("must not contain merge commits") ||
-           e.message.match?(/required status check/i) ||
-           e.message.match?(/cannot force-push to this branch/i)
-          raise BranchProtected
-        end
+        raise BranchProtected if BRANCH_PROTECTION_ERROR_MESSAGES.any? { |msg| e.message.match?(msg) }
```

Comment:
🔎 What do you think of including the message in `BranchProtected`? I think there's a strong value add since a couple of the messages we're handling are related to protections that are more recent (merge queue usage, repository rules).

```suggestion
        raise BranchProtected, e.message if BRANCH_PROTECTION_ERROR_MESSAGES.any? { |msg| e.message.match?(msg) }
```

Reply:
In general I'm a fan of that, but here specifically we just use it as a signal in the code that runs this (api in our case) to perform a different action. We're essentially using errors for control flow 🙄 so I'm not sure we'd benefit?

Reply:
I trust your understanding 😄 If the running code does anything with the signal (reporting to an error tracker), it'd be great to have the extra context. If not, no need.

Reply:
It does not, but I suppose there's no harm in adding it either